draft-ietf-rtcweb-jsep.xml

```diff
@@ -1977,13 +1977,205 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
+              <list style="symbols">
+                <t>If the type is "offer", the PeerConnection state MUST be
+                      either "stable" or "have-remote-offer".</t>
+                <t>If the type is "pranswer" or "answer", the PeerConnection
+                      state MUST be either "have-local-offer" or
+                      "have-remote-pranswer".</t>
+              </list>
+            </t>
+            <t>If the type is not correct for the current state, processing MUST
+            stop and an error MUST be returned.</t>
+            <t>Next, the SessionDescription is parsed into a data structure,
+              as described in the
+              <xref target="sec.parsing-a-desc"/> section below.
+            </t>
+            <t>Finally, the parsed SessionDescription is applied as described in
+              the <xref target="sec.applying-a-remote-desc"/> section below.</t>
+          </list>
+        </t>
+      </section>
 
-      <section title="Applying a Local Description" anchor="sec.applying-a-local-descript"></section>
+      <section title="Parsing a Session Description" anchor="sec.parsing-a-desc">
+        <t>When a SessionDescription of any type is supplied to
+        setLocal/RemoteDescription, the implementation must parse it and
+        reject it if it is invalid. The exact details of this
+        process are explained below.</t>
+        <t>The SDP contained in the session description object consists of a
+        sequence of text lines, each containing a key=value expression.
+        The SDP is read,
+        line-by-line, and converted to a data structure that contains the
+        deserialized information. However, SDP supports various fields that are not relevant to JSEP applications, and so these fields can be ignored.
+        For each line, the implementation will first ensure it is syntactically
+        correct, and then either parse the data to store for later use,
+        or discard the line and move on. If the line is not well-formed, or
+        cannot be parsed as described, the specified the parser MUST stop with an error and reject the session decription. This ensures that implementations do not accidentally misinterpret ambiguous SDP.</t>
+
+        <section title="Session-Level Parsing">
+          <t>First, the session-level lines are checked and parsed. These
+            lines MUST occur in a specific order, and with a specific syntax, as defined in <xref target="RFC4566"/>, Section 5. </t>
+          <t>For non-attribute (a=) lines, their sequencing and syntax are checked,
+            as mentioned above, but all values MUST be discarded. For example,
+            because the use of ICE is mandatory, the c= line is superfluous and its value MUST be discarded; similarly, because session-level bandwidth is ambiguous when multiple media streams are present, a b= line at session
+            level is not useful and its value MUST be discarded.</t>
```

Comment:
I was originally just going to have these as examples, but I think that specifically saying b=/c= etc are ignored will be necessary to have consistency. Haven't yet done that though.